lib/charms/alertmanager_k8s/v0/alertmanager_remote_configurer.py

```diff
@@ -0,0 +1,1888 @@
+    """API that manages a provided `alertmanager_remote_configurer` relation.
+
+    The `AlertmanagerRemoteConfigurerProvider` is intended to be used by charms whose workloads
+    need to receive data from other charms' workloads over the `alertmanager_remote_configurer`
+    interface.
+
+    The `AlertmanagerRemoteConfigurerProvider` object can be instantiated as follows in your charm:
+
+    ```
+    from charms.alertmanager_k8s.v0.alertmanager_remote_configurer import (
+        AlertmanagerRemoteConfigurerProvider,
+    )
+
+    def __init__(self, *args):
+        ...
+        self.remote_configurer_provider = AlertmanagerRemoteConfigurerProvider(self)
+        ...
+    ```
+
+    The `AlertmanagerRemoteConfigurerProvider` assumes that, in the `metadata.yaml` of your charm,
+    you declare a provided relation as follows:
+
+    ```
+    provides:
+        remote-configurer:  # Relation name
+            interface: alertmanager_remote_configurer  # Relation interface
+            limit: 1
+    ```
+
+    When `remote-configurer` is created, `AlertmanagerRemoteConfigurerProvider` reads the current
+    configuration of the Alertmanager from the specified configuration file and pushes the content
+    to the relation data bag. This initial configuration can then be used by the consumer
+    of the relation.
+    The `AlertmanagerRemoteConfigurerProvider` provides 2 public methods for accessing the data
+    from the relation data bag - `config` and `templates`. Typical usage of these methods in the
+    provider charm would look something like:
+
+    ```
+    def get_config(self, *args):
+        ...
+        alertmanager_config = self.remote_configurer_provider.config()
+        ...
+        self.container.push("/alertmanager/config/file.yml", alertmanager_config)
+        ...
+    ```
+
+    ```
+    def get_templates(self, *args):
+        ...
+        alertmanager_templates = self.remote_configurer_provider.templates()
+        ...
+        self.container.push("/alertmanager/templates/file.tmpl", alertmanager_templates)
+        ...
+    ```
+
+    Separation of the main configuration and the templates is dictated by the assumption that
+    the default provider of the `alertmanager_remote_configurer` relation will be
+    `alertmanager-k8s` charm, which requires such separation.
+
+    The `AlertmanagerRemoteConfigurerProvider` also provides a JSON Schema of the Alertmanager
+    configuration. Before returning configuration to the provider charm,
+    `AlertmanagerRemoteConfigurerProvider` validates it. Configuration that passed the validation
+    is returned to the provider charm. Invalid configurations are discarded (empty dict
+    is returned to the provider charm) and the relevant error message is logged.
+    """
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        relation_name: str = DEFAULT_RELATION_NAME,
+        config_file_path: str = DEFAULT_ALERTMANAGER_CONFIG_FILE_PATH,
```

Comment:
Feels like the library shouldn't communicate with what's on disk but rather pass the already read content. What do you think?

Reply:
Yes, that makes sense.
Changed.